testsuite/integration-tests/pom.xml

```diff
@@ -203,6 +203,167 @@
                 <microprofile.excluded.groups>org.jboss.resteasy.category.ExpectedFailingWithStandaloneMicroprofileConfiguration,org.jboss.resteasy.category.ExpectedFailingBecauseOfSmallRyeMicroprofileOpenApi11</microprofile.excluded.groups>
             </properties>
         </profile>
+        
+        <profile>
+            <id>bootablejar.profile</id>
+            <activation>
+                <property>
+                    <name>ts.bootable</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <executions>
+                            <execution>
```

Comment:
I'm thinking about `unpack` and `unpack-resteasy` executions. Why are they needed? According to the Readme file, `-Dserver.home=not-needed-for-bootable.jar` should be used. In that case, download profile won't be activated, so there is no need to replace `unpack` and `unpack-resteasy` executions.

If `-Dserver.home=not-needed-for-bootable.jar` won't be used, download profile will be activated together with `bootablejar.profile`. In this case, there are two active profiles with the executions of the same names. And AFAIR there is no guarantee from maven side which profile will be executed.

Reply:
Thanks for this comment, this is interesting. So there's no guarantee that - when explicitly using `-Dts.bootable` the executions here will be used instead of those defined in the download profile?
What would you suggest then? Following up on your reasoning, then we know for sure that those in the download profile will be executed if `-Dserver.home=not-needed-for-bootable.ja` is passed, once these are removed.
And being the precondition the same I think this last one is a worse scenario. WDYT?
Anyway, I agree with you and I'd like to have a clean solution here.

Reply:
> So there's no guarantee that - when explicitly using -Dts.bootable the executions here will be used instead of those defined in the download profile?

AFAIK you can define execution in default `<build>` once at most. And you can override it in one profile at most. There is nothing like profile orders in maven AFAIK. So if you don't define execution in default `<build>`, but you define it two times differently in two different active profiles, you don't have any guarantee, which definition will be used (@asoldano @ronsigal feel free to correct me, if I'm wrong).

> those in the download profile will be executed if `-Dserver.home=not-needed-for-bootable.jar` is passed

Actually, download profile is activated like this:

```
          <activation>
            <property>
              <name>!server.home</name>
            </property>
          </activation>
```

So if `-Dserver.home=not-needed-for-bootable.jar` is passed, download profile is not activated (see `!` in `name` element), which is something that we want here.

> I'd like to have a clean solution here.

`-Dserver.home=not-needed-for-bootable.jar` is obviously not super-clean solution. But it is a solution and it is deterministic. I don't see any other simple solution without completely changing the way how maven profiles are constructed now in this repo. @asoldano / @ronsigal Let us know if you have some better idea here ...

Reply:
Thanks @marekkopecky - at the moment I am focusing on the fact that the we need those executions to be skipped.
As you say, the current implementation only executes those when `!server.home`, i.e. when the download profile is activated.
And this is deterministic, so I agree with you that this could be the most robust solution.
I am going to push changes, let's see how CI checks go.

Reply:
Can we resolve this now, @marekkopecky? WDYT?

Reply:
Yes, thanks!